### PR TITLE
Add more guest functionality added to the template

### PR DESCRIPTION
**Pull request by meenakshiAS**
I have created a new folder (js) in the static forlder.
The "js" folder has file name called "add_guests.js" that contains the javascript code for add more guest functionality
Also made some changes in the "reservation.html" file

### CODE REVIEW

File: reservations/views.py [hunk 1]
```diff
@@ -1,6 +1,83 @@
...
+        "website/reservation.html",
+        {
+            "reservation_form": reservation_form,
+            "guest_formset": guest_formset,
+        },
+    )
+
+
+def reservation_success(request):
+    return render(request, "website/reservation_success.html")
```
Comment (amakarudze) [inline reservations/views.py:83]:
```suggestion

```
This is not necessary remove it.

File: reservations/urls.py [hunk 2]
```diff
@@ -1,3 +1,11 @@
+from django.urls import path
+
+from . import views
+from .views import staff_sign_up_for_guest, make_reservation
+
 app_name = "reservations"  # This defines the namespace
 
-urlpatterns = []
+urlpatterns = [
+    path('sign-up-for-guest/', views.staff_sign_up_for_guest, name='sign-up-for-guest'),
```
Comment (amakarudze) [inline reservations/urls.py:9]:
```suggestion
   
```

Let's delete this and go with one view.

File: reservations/urls.py [hunk 3]
```diff
@@ -1,3 +1,11 @@
+from django.urls import path
+
+from . import views
+from .views import staff_sign_up_for_guest, make_reservation
```
Comment (amakarudze) [inline reservations/urls.py:4]:
```suggestion

```
You are importing views twice, not necessary and not good at all.

File: reservations/views.py [hunk 4]
```diff
@@ -1,6 +1,83 @@
...
+
+from accounts.models import UserProfile, User
 
 
 def dashboard(request):
     """Dashboard page for the staff reservation website"""
     return render(request, "reservations/index.html", {"title": "Home"})
+
+
+def staff_sign_up_for_guest(request):
```
Comment (amakarudze) [inline reservations/views.py:16]:
```suggestion
def add_reservation(request):

```
Let's rename this view so it makes sense.

File: reservations/views.py (hunk 1 above)
Comment (amakarudze) [inline reservations/views.py:83]:
```suggestion
    form = StaffReservationForm()
    if request.method == "POST":
        form = AddReservationForm(request.POST)
        guest_formset = GuestFormSet(request.POST)
        if form.is_valid():
            user = User.objects.create_user(
                email=form.cleaned_data.get("email"),
                first_name=form.cleaned_data.get("first_name"),
                last_name=form.cleaned_data.get("last_name"),
                password=form.cleaned_data.get("password"),
            )
            UserProfile.objects.create(
            user=user,
            dob=form.cleaned_data.get("dob"),
            address=form.cleaned_data.get("address"),
            city=form.cleaned_data.get("city"),
            postal_code=form.cleaned_data.get("postal_code"),
            state=form.cleaned_data.get("state"),
            country=form.cleaned_data.get("country"),
            phone_number=form.cleaned_data.get("phone_number"),
            )
            reservation = Reservation.objects.create(
            user=user,
            number_of_adults=form.cleaned_data.get("number_of_adults")
            number_of_children=form.cleaned_data.get("number_of_children")
            rooms=form.cleaned_data.get("rooms")
            events=form.cleaned_data.get("event")
            check_in_date=form.cleaned_data.get("check_in_date")
            check_out_date=form.cleaned_data.get("check_out_date")
              )
        if guest_formset.is_valid():
             for guest_form in guest_formset:
                if guest_form.cleaned_data:  
                    guest = guest_form.save(commit=False)
                    guest.reservation = reservation
                    guest.save()

            messages.success(request, "New reservation added successfully")
            return redirect("reservations:reservations_list")
        
    else:
        form = AddReservationForm()

    return render(request, "reservations/reservation.html", {"form": form, "guest_formset": guest_formset})

```

Let's go with one view like this.

**Review by amakarudze — CHANGES_REQUESTED** [review]
Hey @meenakshiAS. Thanks for your work on this. May you kindly address the changes I have suggested. Thanks. Also update your test and fixtures accordingly.

**Comment by meenakshiAS** [comment]
> Hey @meenakshiAS. Thanks for your work on this. May you kindly address the changes I have suggested. Thanks. Also update your test and fixtures accordingly.

ok

**Comment by amakarudze** [comment]
Closing this one again to avoid confusion.

**Pull request closed without merging**